Backend/AMQPBackendDispatcher.php

```diff
@@ -55,25 +57,39 @@ public function __construct(array $settings, array $queues, $defaultQueue, array
     }
 
     /**
-     * @return AMQPChannel
+     * @return AmqpContext
      */
-    public function getChannel()
+    public function getContext()
     {
-        if (!$this->channel) {
-            $this->connection = new AMQPConnection(
-                $this->settings['host'],
-                $this->settings['port'],
-                $this->settings['user'],
-                $this->settings['pass'],
-                $this->settings['vhost']
-            );
+        if (!$this->context) {
+            if (false == array_key_exists('factory_class', $this->settings)) {
```

Comment:
Please use `!`, the return value is already a boolean, no need to compare it to another boolean